libs/designsystem/src/lib/components/calendar/calendar.component.spec.ts

```diff
@@ -0,0 +1,237 @@
+    const today = localMidnightDate('1997-08-28');
+    spectator.setInput('disabledDates', [today]);
+    spectator.setInput('todayDate', today);
+    spectator.setInput('alwaysEnableToday', true);
+    spectator.setInput('selectedDate', localMidnightDate('1997-08-29'));
+
+    const captured = captureDateChangeEvents();
+
+    clickDayOfMonth(28); // August 28th is was disabled but we override
+    expect(captured.event).toEqual(today);
+  });
+
+  it('should emit dateChange event as UTC midnights when when timezone is set to UTC', () => {
+    spectator.setInput('timezone', 'UTC');
+    spectator.setInput('selectedDate', utcMidnightDate('1997-08-29'));
+
+    const captured = captureDateChangeEvents();
+
+    clickDayOfMonth(14);
+    expect(captured.event).toEqual(utcMidnightDate('1997-08-14'));
+  });
+
+  it('should be tolerant of date input (selectedDate, todayDate, minDate, maxDate, and disabledDates) as both UTC midnight and local time midnight', () => {
+    const localDate = localMidnightDate('1997-08-29');
+    const utcDate = utcMidnightDate('1997-08-29');
+
+    spectator.setInput('selectedDate', localDate);
+    expect(spectator.component.selectedDate).toEqual(localDate);
+    spectator.setInput('selectedDate', utcDate);
+    expect(spectator.component.selectedDate).toEqual(localDate);
+
+    spectator.setInput('todayDate', localDate);
+    expect(spectator.component.todayDate).toEqual(localDate);
+    spectator.setInput('todayDate', utcDate);
+    expect(spectator.component.todayDate).toEqual(localDate);
+
+    spectator.setInput('minDate', localDate);
+    expect(spectator.component.minDate).toEqual(localDate);
+    spectator.setInput('minDate', utcDate);
+    expect(spectator.component.minDate).toEqual(localDate);
+
+    spectator.setInput('maxDate', localDate);
+    expect(spectator.component.maxDate).toEqual(localDate);
+    spectator.setInput('maxDate', utcDate);
+    expect(spectator.component.maxDate).toEqual(localDate);
+
+    const localDates = ['1997-08-29', '1997-08-30'].map(localMidnightDate);
+    const utcDates = ['1997-08-29', '1997-08-30'].map(utcMidnightDate);
+
+    spectator.setInput('disabledDates', localDates);
+    expect(spectator.component.disabledDates).toEqual(localDates);
+    spectator.setInput('disabledDates', utcDates);
+    expect(spectator.component.disabledDates).toEqual(localDates);
+  });
+
+  // constants and utility functions
+
+  const SEL_NAV_BACK = '.previous-month-button';
+  const SEL_NAV_FORWARD = '.next-month-button';
+
+  function localMidnightDate(yyyyMMdd) {
+    return moment(yyyyMMdd).toDate();
+  }
+
+  function utcMidnightDate(yyyyMMdd) {
+    return moment.utc(yyyyMMdd).toDate();
+  }
+
+  function clickDayOfMonth(dateOneIndexed: number) {
+    spectator.click(spectator.queryAll('.day.current-month')[dateOneIndexed - 1]);
+  }
+
+  function trimmedTexts(selector: string) {
+    return spectator.queryAll(selector).map((_) => _.textContent.trim());
```

Comment:
Consider using `[HTMLElement.innerText](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText)` instead:
> Basically, innerText is aware of the rendered appearance of text, while textContent is not
```suggestion
    return spectator.queryAll<HTMLElement>(selector).map((_) => _.innerText);
```